pkg/query-service/app/integrations/messagingQueues/kafka/translator.go

```diff
@@ -35,6 +35,139 @@ func BuildQueryRangeParams(messagingQueue *MessagingQueue, queryContext string)
 	return queryRangeParams, nil
 }
 
+func buildClickHouseQueryNetwork(messagingQueue *MessagingQueue, queueType string) (*v3.ClickHouseQuery, error) {
+	start := messagingQueue.Start
+	end := messagingQueue.End
+	consumerGroup, ok := messagingQueue.Variables["consumer_group"]
+	if !ok {
+		return nil, fmt.Errorf("consumer_group not found in the request")
+	}
+
+	query := generateNetworkLatencyThroughputSQL(start, end, consumerGroup, queueType)
+
+	return &v3.ClickHouseQuery{
+		Query: query,
+	}, nil
+}
+
+func buildBuilderQueriesNetwork(attributeCache []Clients) (map[string]*v3.BuilderQuery, error) {
+	bq := make(map[string]*v3.BuilderQuery)
+
+	for i, instanceInfo := range attributeCache {
+		queryName := fmt.Sprintf("latency_%d", i)
+		chq := &v3.BuilderQuery{
+			QueryName:    queryName,
+			StepInterval: defaultStepInterval,
+			DataSource:   v3.DataSourceMetrics,
+			AggregateAttribute: v3.AttributeKey{
+				Key: "kafka_consumer_fetch_latency_avg",
+			},
+			AggregateOperator: v3.AggregateOperatorAvg,
+			Temporality:       v3.Unspecified,
+			TimeAggregation:   v3.TimeAggregationAvg,
+			SpaceAggregation:  v3.SpaceAggregationAvg,
+			Filters: &v3.FilterSet{
+				Operator: "AND",
+				Items: []v3.FilterItem{
+					{
+						Key: v3.AttributeKey{
+							Key:      "service_name",
+							Type:     v3.AttributeKeyTypeTag,
+							DataType: v3.AttributeKeyDataTypeString,
+						},
+						Operator: v3.FilterOperatorEqual,
+						Value:    instanceInfo.ServiceName,
+					},
+					{
+						Key: v3.AttributeKey{
+							Key:      "client_id",
+							Type:     v3.AttributeKeyTypeTag,
+							DataType: v3.AttributeKeyDataTypeString,
+						},
+						Operator: v3.FilterOperatorEqual,
+						Value:    instanceInfo.ClientID,
+					},
+					{
+						Key: v3.AttributeKey{
+							Key:      "service_instance_id",
+							Type:     v3.AttributeKeyTypeTag,
+							DataType: v3.AttributeKeyDataTypeString,
+						},
+						Operator: v3.FilterOperatorEqual,
+						Value:    instanceInfo.ServiceInstanceID,
+					},
```

Comment:
I realized, instead of creating one query for each combination of (client_id, service_name, service_instance_id), we should just create one query.

```
					{
						Key: v3.AttributeKey{
							Key:      "service_name",
							Type:     v3.AttributeKeyTypeTag,
							DataType: v3.AttributeKeyDataTypeString,
						},
						Operator: v3.FilterOperatorIn,
						Value:   [...serviceNames],
					},

``` 

Because there is group by `service_name, client_id, service_instance_id` which means it going to give result for them separately anyway with one query. This will make difference when the number of unique combinations are more than few because it would run as many queries as service_instances_id which can potentiall go into tens of ids.